src/pages/Profile.js

```diff
@@ -1,22 +1,77 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { useContext, useEffect, useState } from 'react';
+import {Link, useHistory} from 'react-router-dom';
+import { AuthContext } from '../context/AuthContext';
+import axios from 'axios';
+import jwtDecode from "jwt-decode";
+import checkValidationOfJWT from "../helpers/checkValidationOfJWT";
 
 function Profile() {
-  return (
-    <>
-      <h1>Profielpagina</h1>
-      <section>
-        <h2>Gegevens</h2>
-        <p><strong>Gebruikersnaam:</strong> hardcoded-test</p>
-        <p><strong>Email:</strong> hardcoded@test.com</p>
-      </section>
-      <section>
-        <h2>Strikt geheime profiel-content</h2>
-        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
-      </section>
-      <p>Terug naar de <Link to="/">Homepagina</Link></p>
-    </>
-  );
+    const [profileData, setProfileData] = useState({});
```

Comment:
Ik zou de default waarde `undefined` maken, zodat je een beter (te controleren) onderscheid hebt tussen of het al wel of niet is ingeladen
```suggestion
    const [profileData, setProfileData] = useState();
```